src/Func.h

```diff
@@ -1351,12 +1351,20 @@ class Func {
     }
     // @}
 
-    /** Schedule for execution using GLSL. Conceptually, this is similar to
-     * parallelization over 'x' and 'y' (since GLSL shaders compute individual
-     * output pixels in parallel) and vectorization over 'c' (since GLSL
-     * implicitly vectorizes the color channel). */
+    /** Schedule for execution using coordinate-based hardware api.
+     * GLSL and Renderscript are examples of those. Conceptually, this is
+     * similar to parallelization over 'x' and 'y' (since GLSL shaders compute
+     * individual output pixels in parallel) and vectorization over 'c'
+     * (since GLSL/RS implicitly vectorizes the color channel). */
+    EXPORT Func &image(Var x, Var y, Var c, DeviceAPI device_api);
+
+    /** Schedule for execution as GLSL kernel. */
     EXPORT Func &glsl(Var x, Var y, Var c);
 
+    /** Schedule for execution as Renderscript kernel. */
```

Comment:
I'd rather not introduce explicit scheduling of this form. Rather one should call image above with the approrpiate device_api parameter. We need to leave glsl in for backward compatibility, but the preferred style should be to call image. (The reason is each API that gets added would need its own scheduling directive.)

I also thing "image" is the wrong word. I'd say "shader" is probably better